data/start-cosmic

```diff
@@ -14,13 +16,23 @@ for unit in $(systemctl --user --no-legend --state=failed --plain list-units | c
     done
 done
 
-# /etc/profile contains a lot of important environment variables
-source /etc/profile
+# use the user's preferred shell to acquire environment variables
+# see: https://github.com/pop-os/cosmic-session/issues/23
+if [ -n "${SHELL}" ]; then
+    # --in-login-shell: our flag to indicate that we don't need to recurse any further
+    if [ "${1}" != "--in-login-shell" ]; then
+        # `exec -l`: like `login`, prefixes $SHELL with a hyphen to start a login shell
+        exec bash -c "exec -l '${SHELL}' -c '${0} --in-login-shell'"
+    fi
+else
```

Comment:
I did consider completely dropping the old `source ...` logic here, making the assumption that SHELL=/usr/bin/bash is very likely to be the default for new user accounts on most distributions :shrug:

Reply:
If we switch to a login shell, we should remove the source logic, given other shell may not be able to read /etc/profile.

If the user changes it's shell, they should get what they want.